api/python/tests/integration/test_packages.py

```diff
@@ -863,6 +863,15 @@ def test_top_hash_stable(self):
         assert pkg.top_hash == top_hash, \
             "Unexpected top_hash for {}/packages/.quilt/packages/{}".format(registry, top_hash)
 
+    def test_top_hash_stable_dir_meta(self):
+        registry = DATA_DIR.as_posix()
+        top_hash = '20de5433549a4db332a11d8d64b934a82bdea8f144b4aecd901e7d4134f8e733'
```

Comment:
this is fine but the unit test will break every time someone modifies `/data`. So I'd use set_dir here, get the `.top_hash` and then assert that the hash changes after calling `.set_meta`.

then maybe patch `get_top_hash_parts` (which doesn't have any test coverage) and return whatever you want so you can check against a stable hash?

Reply:
I merged the new test with `test_top_hash_stable()`. We could completely inline the [test manifest](https://github.com/quiltdata/quilt/blob/master/api/python/tests/integration/data/.quilt/packages/20de5433549a4db332a11d8d64b934a82bdea8f144b4aecd901e7d4134f8e733) in the test if it makes sense to you.